mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxConstants.java

```diff
@@ -46,4 +46,6 @@
     /** The maximum number of annotations on a mailbox */
     int DEFAULT_LIMIT_ANNOTATIONS_ON_MAILBOX = 10;
 
+    /** The limitation of mailbox's name */
+    int DEFAULT_LIMIT_MAILBOX_NAME_SIZE = 256;
```

Comment:
Then it will have problem with ByteBuffer, we just allocate 16384 for reading input, if we increase the big data of mailbox name, than we need to increase allocate value.
Could you please take a look my last commit?